src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs

```diff
@@ -1672,5 +1672,536 @@ .maxstack  1
 }");
         }
 
+        [Fact]
+        public void RefReturnConditionalAccess01()
+        {
+            var text = @"
+    using System;
+ 
+    class Program
+    {
+        class C1<T> where T : IDisposable
+        {
+            T inst = default(T);
+
+            public ref T GetDisposable()
+            {
+                return ref inst;
+            }
+
+            public void Test()
+            {
+                GetDisposable().Dispose();
+                System.Console.Write(inst.ToString());
+
+                GetDisposable()?.Dispose();
```

Comment:
I will check. We should have that covered.

Tests for that would need to go into master though, since they would not require byref returns. The scenarios would be applicable to regular variables or byref parameters just the same.